corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -417,18 +424,22 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         submitAction: function (e) {
             e.preventDefault();
             FormplayerFrontend.trigger('clearNotifications', errorHTML, true);
-
             var invalidFields = [];
             this.children.each(function (childView) {
                 if (!childView.isValid()) {
                     invalidFields.push(childView.model.get('text'));
+                    FormplayerFrontend.trigger('showInputError', gettext(childView.model.get("error")) ||
+                    gettext("Please enter a value for this field."), childView);
                 }
             });
 
             if (invalidFields.length) {
-                var errorHTML = "Please enter values for the following fields:";
-                errorHTML += "<ul>" + _.map(invalidFields, function (f) { return "<li>" + f + "</li>"; }).join("") + "</ul>";
+                var errorHTML = "Please check the following fields:";
+                errorHTML += "<ul>" + _.map(invalidFields, function (f) {
+                    return "<li>" + f + "</li>";
```

Comment:
Realize you just moved this code, but this should use `DOMPurify.sanitize` on `f` since it's user input that's going to be displayed as HTML (unless it's sanitized elsewhere).

Reply:
As far as I can tell it's not sanitized anywhere else. I'll add it here.

Reply:
cool, thank you for investigating

Reply:
...actually, after thinking for 3 more seconds, I think this should still be sanitized. It doesn't come from the web apps user, but it does come from **a** user, the app builder, and we typically sanitize those, too.